node.js/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -3,13 +3,73 @@ const router = express.Router();
 
 const meals = require("./../data/meals.json");
 
-router.get("/", async (request, response) => {
+router.get("/:id", async (request, response) => {
+  console.log(meals);
   try {
-    console.log(meals);
-    console.log("in /api/meals");
+    const mealId = parseInt(request.params.id);
+
+    if (isNaN(mealId)) {
+      response.status(404).json({ error: "Id must be integers." })
+      return;
+    }
+
+    if (mealId <= meals.length) {
+      const mealById = meals.filter((meal) => mealId == meal.id);
+      console.log(mealById);
+      response.json(mealById[0]);
+
+    } else {
+      response.send({});
```

Comment:
response status(404) ? instead of empty json ?

Reply:
Sure, Thanks.

Reply:
Done. Thanks for the feedback.